### ISSUE TITLE
TaskQueue<Success> async execution is not serial

### ISSUE THREAD

**Issue by MuniekMg** [aws-amplify/amplify-swift#3556]
### Describe the bug

While working on other PR I found out that one unit test sometimes fails. This test expected correct order of function executions. 

TaskQueue<Success> has async(...) function which should schedule task and execute this task in serial manner but it doesn't.

### Steps To Reproduce

```swift
1. Add 1000 task to TaskQueue<Void>.async { print(i) }
2. Tasks won't be executed in order
```

### Expected behavior

Tasks are being executed in order

### Amplify Framework Version

2.27.0

### Amplify Categories

DataStore

### Dependency manager

Swift PM

### Swift version

5

### CLI version

10.3.1

### Xcode version

Version 15.2 (15C500b)

### Is this a regression?

No

### Platforms

iOS

### OS Version

iOS 17

### Device

iPad 10th gen

**Comment by 5d** [comment]
Hi @MuniekMg ,

Thanks for reporting the issue and submitting the proposed PR. Our team will investigate and review the PR.

**Comment by 5d** [comment]
The PR has been released in version 2.29.2. We are grateful for your valuable contribution.
